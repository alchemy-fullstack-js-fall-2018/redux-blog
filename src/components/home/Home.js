import React, { Fragment, PureComponent } from 'react';
import PropTypes from 'prop-types';
import Posts from '../posts/Posts';

export default class Home extends PureComponent {
  static propTypes = {
    fetchPosts: PropTypes.func.isRequired,
    longestPosts: PropTypes.array.isRequired,
  };

  render() {
    const { longestPosts, fetchPosts } = this.props;

    if(!fetchPosts || !longestPosts) return null;
    
    return (
      <Fragment>
        <h2>Welcome Home</h2>
        <p>
          Lorem ipsum dolor Bob Ross... Let’s make a happy little mountain now. You have to make almighty decisions when you’re the creator. Let’s make a nice big leafy tree. Remember how free clouds are. They just lay around in the sky all day long.
        </p>
        <Posts posts={longestPosts} fetchPosts={fetchPosts} />
      </Fragment>
    );
  }
}
