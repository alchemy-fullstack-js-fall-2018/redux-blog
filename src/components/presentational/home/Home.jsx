import React from 'react';
import PostListLongest from '../../container/posts/PostListLongest';

const Home = () => {

  console.log(PostListLongest);



  return (
    <section>
      <h2>Top Posts</h2>
      {PostListLongest}
    </section>
  );
};

export default Home;
