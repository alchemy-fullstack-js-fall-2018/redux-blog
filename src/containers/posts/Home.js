import { connect } from 'react-redux';
import Home from '../../components/home/Home';
import { getLongestPosts } from '../../selectors/posts';
import { fetchPosts } from '../../actions/posts';

const mapStateToProps = state => ({
  longestPosts: getLongestPosts(state)
});

const mapDispatchToProps = dispatch => ({
  fetchPosts() {
    dispatch(fetchPosts());
  }
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Home);
