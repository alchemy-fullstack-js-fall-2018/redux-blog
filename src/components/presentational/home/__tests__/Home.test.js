import React from 'react';
import { shallow } from 'enzyme';
import Home from '../Home.jsx';

jest.mock('../../../../routes');

describe('<Home />', () => {
  test('renders', () => {
    const wrapper = shallow(<Home />);
    expect(wrapper).toMatchSnapshot();
  });
});
