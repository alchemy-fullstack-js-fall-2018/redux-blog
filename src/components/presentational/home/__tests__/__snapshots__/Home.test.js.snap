// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> renders 1`] = `
<section>
  <h2>
    This is a blog all about awesome Redux is
  </h2>
  <p>
    It's obviously the best state management system
  </p>
</section>
`;
